lib/spack/spack/config.py

```diff
@@ -391,63 +391,66 @@ class Configuration(object):
     This class makes it easy to add a new scope on top of an existing one.
     """
 
-    def __init__(self, *scopes):
+    # convert to typing.OrderedDict when we drop 3.6, or OrderedDict when we reach 3.9
+    scopes: Dict[str, ConfigScope]
+
+    def __init__(self, *scopes: ConfigScope):
```

Comment:
Should this be `List[ConfigScope]`?

Reply:
No: https://peps.python.org/pep-0484/#arbitrary-argument-lists-and-default-argument-values

Reply:
Oh right